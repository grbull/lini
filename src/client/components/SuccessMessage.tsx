import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

interface Props {
  children: string | ReactNode;
  className?: string;
  illustration?: boolean;
}

export function SuccessMessage({
  children,
  className,
  illustration,
}: Props): ReactElement {
  return (
    <div
      className={cn(
        'px-2',
        'py-2.5',
        'bg-green-400',
        'bg-opacity-50',
        'border',
        'border-solid',
        'border-green-400',
        className
      )}
    >
      <div
        className={cn('flex', 'flex-col', 'items-center', 'justify-center', {
          hidden: !illustration,
        })}
      >
        <img
          alt="Happy Illustration"
          className={cn('opacity-80', 'mb-2.5')}
          src="happy.png"
          width={80}
        />
      </div>
      <h2 className={cn('font-bold')}>Success</h2>
      <p>{children}</p>
    </div>
  );
}

SuccessMessage.defaultProps = { className: undefined, illustration: false };
