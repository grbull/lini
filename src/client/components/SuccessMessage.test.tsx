/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { SuccessMessage } from './SuccessMessage';

function setup(withIllustration = false): RenderResult {
  const utils = render(
    <SuccessMessage illustration={withIllustration}>
      Generic children
    </SuccessMessage>
  );

  return { ...utils };
}

describe('SuccessMessage Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with illustration', () => {
    const { asFragment } = setup(true);
    expect(asFragment()).toMatchSnapshot();
  });
});
