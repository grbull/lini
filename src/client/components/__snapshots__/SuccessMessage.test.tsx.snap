// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessage Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-2 py-2.5 bg-green-400 bg-opacity-50 border border-solid border-green-400"
  >
    <div
      class="flex flex-col items-center justify-center hidden"
    >
      <img
        alt="Happy Illustration"
        class="opacity-80 mb-2.5"
        src="happy.png"
        width="80"
      />
    </div>
    <h2
      class="font-bold"
    >
      Success
    </h2>
    <p>
      Generic children
    </p>
  </div>
</DocumentFragment>
`;

exports[`SuccessMessage Component matches the snapshot with illustration 1`] = `
<DocumentFragment>
  <div
    class="px-2 py-2.5 bg-green-400 bg-opacity-50 border border-solid border-green-400"
  >
    <div
      class="flex flex-col items-center justify-center"
    >
      <img
        alt="Happy Illustration"
        class="opacity-80 mb-2.5"
        src="happy.png"
        width="80"
      />
    </div>
    <h2
      class="font-bold"
    >
      Success
    </h2>
    <p>
      Generic children
    </p>
  </div>
</DocumentFragment>
`;
